### PR TITLE
Add CLAUDE.md documentation for AI assistant guidance

### PR DESCRIPTION
**Pull request by tjmgregory**
## Summary
- Added comprehensive CLAUDE.md file to guide AI assistants working with the FIRE project
- Captures project overview, core principles, technical stack, and development guidelines
- Consolidates information from existing documentation, cursor rules, and codebase patterns

## Test plan
- [x] Review CLAUDE.md content for accuracy
- [x] Verify all referenced file paths exist
- [x] Ensure code examples match current implementation patterns
- [x] Check that security guidelines are properly documented

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  }
+  // Process...
+}
+```
+
+### Logging Levels
+- `console.error()`: Actual errors preventing operation
+- `console.warn()`: Potential issues (deprecated features, etc.)
+- `console.info()`: Important business events
+- `console.log()`: Debugging information
```
Comment (tjmgregory) [inline CLAUDE.md:84]: Can you please confirm whether the apps script runtime actually supports these levels of logging? If it doesn't please update this doc and any other docs that mention logging.

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  - Types: `feat`, `fix`, `docs`, `style`, `refactor`, `test`, `chore`
+  - Example: `feat(categorization): add support for Amex transactions`
+  - Example: `fix(normalization): handle edge case in date parsing`
+  - Example: `docs(adr): update ADR-001 with new bank format`
+
+### 3. Planning Requirements
+- Create implementation plans in `docs/plans/active/` using format: `YYYY-MM-DD-HH-MM_description.md`
+- Plans must include: Problem Statement, Related ADRs, Implementation Steps (checklist), Test Cases, Lessons Learned
+- Always request code samples when making plans
+- Move completed plans to `docs/plans/completed/`
```
Comment (tjmgregory) [inline CLAUDE.md:33]: So, plans were a thing just for working with Cursor, and aren't needed for Claude. However, the documentation-first principle for the rest of the docs is still important.

**Comment by tjmgregory** [comment]
## Addressed Review Comments

### Logging Levels Comment
Confirmed that Apps Script does support all console logging methods (console.log, console.error, console.warn, console.info). I've updated the documentation to:
- Add "(Supported in Apps Script)" to the logging levels heading
- Added a note about viewing logs via "View > Logs" in the Apps Script editor
- The codebase already uses these methods extensively as documented in the logging standards

### Planning Requirements Comment  
Updated the documentation to reflect that plans are Cursor-specific:
- Removed the "Planning Requirements" section
- Replaced it with "Documentation Standards" that maintains the documentation-first principle
- Updated references to plans directory to indicate they are historical/Cursor-specific
- Removed plans from the documentation locations list
